### DISCUSSION TITLE
Help tweak the programme.

### DISCUSSION THREAD

**Discussion by HenryDu8133** [cc-tweaked/CC-Tweaked#1904] [Q&A] [answered]
Hi there.Recently I've been working on some CCyoucube code, and now I've written a fast audio playback application called ‘Play.lua’.Here is the code for it
```
local req = ""
while true to
       if req ~= " " then shell.run("youcube.lua --nv -V 300 " .. req
       else shell.run("youcube.lua --nv -V 300 ")
       end

       req = " "
       print()
       for i = 1 , 20 do term.write("=") end
       print()
       req = read()
end
```
It's based on youcube's code
[youcube.lua](https://github.com/CC-YouCube/client/blob/main/src/youcube.lua)

It is now possible to play audio by typing ‘Play’
<img width="928" alt="屏幕截图 2024-07-19 133326" src="https://github.com/user-attachments/assets/4fe8c6f7-f953-4ac9-a817-9979934cb437">

Now I want it to get the link directly after ‘Play’ and start playing. No need to enter the link again
Like this:
<img width="427" alt="屏幕截图 2024-07-19 133535" src="https://github.com/user-attachments/assets/32fc5770-bd06-4fe8-931a-3dc083415d1c">

Could you help me to tweak the programme? Thanks !

**Comment by fatboychummy** [comment]
To get a list of arguments, use `...`

```lua
local arg1, arg2, arg3 = ...

-- or, in a table
local args = {...}
```

This can be used inside functions as well:
```lua
local function print_separately(...)
  local args = {...}
  for i = 1, #args do
    print(args[i])
  end
end

print_separately("hello", "world")
```

**Reply by HenryDu8133** [reply]
Give more instructions?

**Reply by fatboychummy** [reply]
```lua
local req = ...
shell.run("youcube.lua --nv -V 300", req)
```

Then when you do `Play https://blablabla.bla` in the shell, it will pass `"https://blablabla.bla"` as `req`.

**Reply by fatboychummy** [reply, marked as answer]
Or alternatively, just pass `...` directly to `shell.run`
```lua
shell.run("youcube.lua --nv -V 300", ...)
```
